src/Types/ContractName.cs

```diff
@@ -36,6 +48,59 @@ public static bool TryParse(string name, out (ContractName? ContractName, Valida
         return validate;
     }
 
+    /// <summary>
+    /// Copies the init name to a byte array which has the length preprended.
+    /// </summary>
+	public byte[] ToBytes()
+    {
+        var bytes = Encoding.ASCII.GetBytes(this.Name);
+
+        using var memoryStream = new MemoryStream((int)this.SerializedLength());
+        memoryStream.Write(Serialization.ToBytes((ushort)bytes.Length));
+        memoryStream.Write(bytes);
+        return memoryStream.ToArray();
+    }
+
+    /// <summary>
+    /// Deserialize an init name from a serialized byte array.
+    /// </summary>
+    /// <param name="bytes">The serialized init name.</param>
+    /// <param name="output">Where to write the result of the operation.</param>
+    public static bool TryDeserial(ReadOnlySpan<byte> bytes, out (ContractName? ContractName, string? Error) output)
+    {
+        if (bytes.Length < MinSerializedLength)
+        {
+            var msg = $"Invalid length of input in `InitName.TryDeserial`. Expected at least {MinSerializedLength}, found {bytes.Length}";
+            output = (null, msg);
+            return false;
+        };
+
+        var sizeRead = BinaryPrimitives.ReadUInt16BigEndian(bytes);
+        var size = sizeRead + MinSerializedLength;
```

Comment:
**Suggestion:** This is not directly tied to the `MinSerializedLength`, but just happens to be the same so I would prefer to avoid using it here
```suggestion
        var size = sizeRead + sizeof(ushort);
```